addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,565 @@
+    type_description = fields.Char(string="Description")
+
+    # Transportation details
+    distance = fields.Integer("Distance", tracking=True)
+    mode = fields.Selection([
+        ("1", "By Road"),
+        ("2", "Rail"),
+        ("3", "Air"),
+        ("4", "Ship or Ship Cum Road/Rail")
+    ], string="Transportation Mode", copy=False, tracking=True, default="1")
+
+    # Vehicle Number and Type required when transportation mode is By Road.
+    vehicle_no = fields.Char("Vehicle Number", copy=False, tracking=True)
+    vehicle_type = fields.Selection([
+        ("R", "Regular"),
+        ("O", "Over Dimensional Cargo")],
+        string="Vehicle Type",
+        compute="_compute_vehicle_type",
+        store=True,
+        copy=False,
+        tracking=True,
+        readonly=False
+    )
+
+    # Document number and date required in case of transportation mode is Rail, Air or Ship.
+    transportation_doc_no = fields.Char(
+        string="Transporter Doc No",
+        copy=False, tracking=True)
+    transportation_doc_date = fields.Date(
+        string="Transporter Doc Date",
+        copy=False,
+        tracking=True)
+
+    transporter_id = fields.Many2one("res.partner", "Transporter", copy=False, tracking=True)
+
+    error_message = fields.Html(readonly=True)
+    blocking_level = fields.Selection([
+        ("warning", "Warning"),
+        ("error", "Error")],
+        string="Blocking Level", readonly=True)
+
+    content = fields.Binary(compute='_compute_content', compute_sudo=True)
+    cancel_reason = fields.Selection(selection=[
+        ("1", "Duplicate"),
+        ("2", "Data Entry Mistake"),
+        ("3", "Order Cancelled"),
+        ("4", "Others"),
+    ], string="Cancel reason", copy=False, tracking=True)
+    cancel_remarks = fields.Char("Cancel remarks", copy=False, tracking=True)
+
+    def _compute_supply_type(self):
+        for ewaybill in self:
+            ewaybill.supply_type = ewaybill.picking_type_code == 'incoming' and 'I' or 'O'
+
+    @api.depends('picking_id')
+    def _compute_document_partners_details(self):
+        for ewaybill in self:
+            picking_id = ewaybill.picking_id
+            if ewaybill.picking_type_code == 'incoming':
+                ewaybill.partner_bill_to_id = picking_id.company_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.picking_type_id.warehouse_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.partner_id
+            else:
+                ewaybill.partner_bill_to_id = picking_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.company_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.picking_type_id.warehouse_id.partner_id
+                if sale_id := ewaybill.picking_id._get_l10n_in_sale():
```

Comment:
If we technically do not have the sale object here, we should avoid to put logic about it here.  Instead of _get_l10n_in_sale, we could just put _l10n_in_get_invoice_partner